model/lifecycle.go

```diff
@@ -432,79 +450,18 @@ func RestartVersions(versionsToRestart []*VersionToRestart, abortInProgress bool
 func RestartBuild(buildId string, taskIds []string, abortInProgress bool, caller string) error {
 	if abortInProgress {
 		// abort in-progress tasks in this build
-		if err := task.AbortTasksForBuild(buildId, taskIds, caller); err != nil {
+		if err := task.AbortAndMarkResetTasksForBuild(buildId, taskIds, caller); err != nil {
```

Comment:
Is there a benefit to marking tasks reset here? We already do that in restartTasks (or is there a way to move that out of that function now?)

Reply:
Only single host TG tasks are marked for reset in `restartTasks` (because we need to wait for the whole group to end before we reset those), so we still need to mark the aborted tasks for reset if we want them to be considered part of the restart.

Reply:
As far as I was aware we don't check ResetWhenFinished for other things. Unless we need this change to make display tasks work?

Reply:
We check it [here](https://gist.github.com/hadjri/e1849bd79ce84b89907a3c3822aaa0cd#file-task_lifecycle-go-L711) when a task is ended by the agent. In this case, the running tasks will get marked `ResetWhenFinished` and `Aborted`, then later the agent will pick up on this and fail the tasks, and this line will hit causing the task to reset automatically.

Reply:
Hm I'm kind of confused about how this works now then. Is this a confirmed bug (that a regular task aborted via this function doesn't get restarted bc resetWhenFinished isn't set) or is this logic handled elsewhere?

Reply:
Yes currently, `RestartBuild` has a bug because the in-progress tasks just get aborted and that's it, there isn't any extra logic handling. The change to also add `ResetWhenFinished` will make it work correctly.

Reply:
Sorry @hadjri I think this comment might have gotten lost since it's an old thread (or let me know if you did address it already)